lib/facter/util/windows/api_types.rb

```diff
@@ -37,6 +37,27 @@ def read_wide_string(char_length, dst_encoding = Encoding::UTF_8)
       str.encode(dst_encoding)
     end
 
+    # @param max_char_length [Integer] Maximum number of wide chars to return (typically excluding NULLs), *not* bytes
+    # @param null_terminator [Symbol] Number of number of null wchar characters, *not* bytes, that determine the end of the string
+    #   null_terminator = :single_null, then the terminating sequence is two bytes of zero.   This is UNIT16 = 0
+    #   null_terminator = :double_null, then the terminating sequence is four bytes of zero.  This is UNIT32 = 0
+    def read_arbitrary_wide_string_up_to(max_char_length = 512, null_terminator = :single_null)

```

Comment:
Yeah, I'm not worried about the little bit of duplication here inside the Facter 2 gem - this is an internal Win32 API helper built around FFI.